ctapipe/image/muon/tests/test_features.py

```diff
@@ -14,6 +19,48 @@ def test_ring_containment():
 
     assert(ring_cont <= 1. and ring_cont >= 0.)
 
+
+def test_ring_completeness():
+
+    angle_ring = np.linspace(0, 2 * mt.pi, 360.)
+    x = np.cos(angle_ring) * u.m
+    y = np.sin(angle_ring) * u.m
+    pe = np.random.uniform(0, 100, len(x))
+    ring_radius = 1. * u.m
+
+    ring_center_x = 0. * u.m
+    ring_center_y = 0. * u.m
+
+    ring_comp = ring_completeness(
+        x, y, pe, ring_radius,
+        ring_center_x, ring_center_y,
+        30, 30)
+
+    assert(ring_comp <= 1. and ring_comp >= 0.)
+
+
+def test_npix_above_threshold():
+    len_array = 100
+    thr = 5.
+    pix = np.random.uniform(0, 100, len_array)
+
+    npix = npix_above_threshold(pix, thr)
+
+    assert((npix >= 0) and (npix <= len_array))
+
+
+def test_npix_composing_ring():
+    len_array = 100
+    pix = np.random.uniform(0, 100, len_array) 
+
+    npix = npix_composing_ring(pix)
+
+    assert((npix >= 0) and (npix <= len_array))
+
+
 if __name__ == '__main__':
 
     test_ring_containment()
+    test_ring_completeness()
+    test_npix_above_threshold()
+    test_npix_composing_ring()
```

Comment:
I see these `__main__` blocks in python-test files here and in digicampipe as well and wonder why people are writing these... 

If one wants to execute only part of the tests, one can do:

    pytest ctapipe/image/muon/tests/test_features.py

for example ...

Reply:
yes, we should remove these main blocks, mainly because the code coverage tools get confused by them and expect you to write unit tests for your unit tests if they are executable in this way.  The correct way to run the tests is with `pytest` (or `python -mpytest <script>`, which is identical).  

This is particularly important for unit tests that use text fixtures (common data items shared among tests, like `test_event` (defined in `conftest.py`) which is a common sample event one can use to test algorithms.

Reply:
I got rid of the main block for the test (I think I included it because I saw it in other test files, but cannot remember which one...)